client/my-sites/plugins/marketplace/marketplace-plugin-details/purchase-area.tsx

```diff
@@ -16,6 +16,7 @@ interface PurchseArea {
 	onAddYoastPremiumToCart: () => void;
 	onNavigateToCheckout: () => void;
 	onNavigateToDomainsSelection: () => void;
+	onRemoveEverythingFromCart: () => void;
```

Comment:
Two things here:

1. The type isn't precise enough to convey the fact that it's an async function. An async function is essentially a function returning `Promise`, so that's what we can do here. You can see how it is done in [the shopping cart type module](https://github.com/Automattic/wp-calypso/blob/trunk/packages/shopping-cart/src/types.ts#L45). Let's update the type declaration for both `onRemoveEverythingFromCart()` and `onAddYoastPremiumToCart()`.
2. Typo: `PurchseArea` should be `PurchaseArea`. It's somehow missed in the last code review, so let's fix it here :)